graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_category_transformation.py

```diff
@@ -157,14 +165,91 @@ def apply(self, input_df: DataFrame) -> DataFrame:
 
         # see get_json_representation() docstring for structure
         self.json_representation = {
-            "string_indexer_labels_array": str_indexer_model.labelsArray,
+            "string_indexer_labels_arrays": str_indexer_model.labelsArray,
             "cols": self.cols,
             "per_col_label_to_one_hot_idx": per_col_label_to_one_hot_idx,
             "transformation_name": self.get_transformation_name(),
         }
 
         return dense_vector_features
 
+    def apply_precomputed_transformation(self, input_df: DataFrame) -> DataFrame:
+
+        # Get JSON representation of categorical transformation
+        labels_arrays: list[list[str]] = self.json_representation["string_indexer_labels_arrays"]
```

Comment:
Is this for task labels? and per_col_label_to_one_hot_idx is for features?
Can you add some comments to clarify it?

Reply:
No the naming comes from Spark and I agree is a bit confusing: https://spark.apache.org/docs/3.5.0/api/python/reference/api/pyspark.ml.feature.StringIndexerModel.html#pyspark.ml.feature.StringIndexerModel.labelsArray

Spark refers to unique values (strings) in a categorical column as "labels". 

I'm adding comments here to clarify.